bundles/org.openhab.core.thing/src/test/java/org/eclipse/smarthome/core/thing/binding/builder/ThingBuilderTest.java

```diff
@@ -18,70 +18,80 @@
-        ThingBuilder thingBuilder = ThingBuilder.create(THING_TYPE_UID, THING_UID);
-        thingBuilder.withChannels(Arrays.asList( //
+        builder.withChannels(Arrays.asList( //
                 ChannelBuilder.create(new ChannelUID(THING_UID, "channel1"), "").build(), //
                 ChannelBuilder.create(new ChannelUID(THING_UID, "channel1"), "").build()));
     }
 
     @Test(expected = IllegalArgumentException.class)
     public void testWithChannels_duplicatesVararg() {
-        ThingBuilder thingBuilder = ThingBuilder.create(THING_TYPE_UID, THING_UID);
-        thingBuilder.withChannels( //
+        builder.withChannels( //
                 ChannelBuilder.create(new ChannelUID(THING_UID, "channel1"), "").build(), //
                 ChannelBuilder.create(new ChannelUID(THING_UID, "channel1"), "").build());
     }
 
     @Test
     public void testWithoutChannel() {
-        ThingBuilder thingBuilder = ThingBuilder.create(THING_TYPE_UID, THING_UID);
-        thingBuilder.withChannels( //
+        builder.withChannels( //
                 ChannelBuilder.create(new ChannelUID(THING_UID, "channel1"), "").build(), //
                 ChannelBuilder.create(new ChannelUID(THING_UID, "channel2"), "").build());
-        thingBuilder.withoutChannel(new ChannelUID(THING_UID, "channel1"));
-        assertThat(thingBuilder.build().getChannels().size(), is(equalTo(1)));
-        assertThat(thingBuilder.build().getChannels().get(0).getUID().getId(), is(equalTo("channel2")));
+        builder.withoutChannel(new ChannelUID(THING_UID, "channel1"));
+        Thing thing = builder.build();
+        assertThat(thing.getChannels().size(), is(equalTo(1)));
+        assertThat(thing.getChannels().get(0).getUID().getId(), is(equalTo("channel2")));
     }
 
     @Test
     public void testWithoutChannel_missing() {
-        ThingBuilder thingBuilder = ThingBuilder.create(THING_TYPE_UID, THING_UID);
-        thingBuilder.withChannels( //
+        builder.withChannels( //
                 ChannelBuilder.create(new ChannelUID(THING_UID, "channel1"), "").build(), //
                 ChannelBuilder.create(new ChannelUID(THING_UID, "channel2"), "").build());
-        thingBuilder.withoutChannel(new ChannelUID(THING_UID, "channel3"));
-        assertThat(thingBuilder.build().getChannels().size(), is(equalTo(2)));
+        builder.withoutChannel(new ChannelUID(THING_UID, "channel3"));
+        assertThat(builder.build().getChannels().size(), is(equalTo(2)));
     }
 
     @Test(expected = IllegalArgumentException.class)
     public void testWithChannel_wrongThing() {
-        ThingBuilder thingBuilder = ThingBuilder.create(THING_TYPE_UID, THING_UID);
-        thingBuilder.withChannel(
+        builder.withChannel(
                 ChannelBuilder.create(new ChannelUID(new ThingUID(THING_TYPE_UID, "wrong"), "channel1"), "").build());
     }
+
+    @Test
+    @Ignore
+    public void subsequentBuildsCreateIndependentThings() {
```

Comment:
Right now this is the pain point. We have to refactor the whole `ThingBuilder` to complete the feature. On the other hand the `ThingBuilder` currently uses an internal instance of the `ThingImpl` instead of properties.